atom/browser/osr/osr_render_widget_host_view.cc

```diff
@@ -909,128 +859,70 @@ void OffScreenRenderWidgetHostView::SetNeedsBeginFrames(
   begin_frame_timer_->SetActive(needs_begin_frames);
 
   if (software_output_device_) {
-    software_output_device_->SetActive(needs_begin_frames && painting_, false);
-  }
-}
-
-void CopyBitmapTo(const SkBitmap& destination,
-                  const SkBitmap& source,
-                  const gfx::Rect& pos) {
-  char* src = static_cast<char*>(source.getPixels());
-  char* dest = static_cast<char*>(destination.getPixels());
-  int pixelsize = source.bytesPerPixel();
-
-  int width =
-      pos.x() + pos.width() <= destination.width()
-          ? pos.width()
-          : pos.width() - ((pos.x() + pos.width()) - destination.width());
-  int height =
-      pos.y() + pos.height() <= destination.height()
-          ? pos.height()
-          : pos.height() - ((pos.y() + pos.height()) - destination.height());
-
-  if (width > 0 && height > 0) {
-    for (int i = 0; i < height; i++) {
-      memcpy(dest + ((pos.y() + i) * destination.width() + pos.x()) * pixelsize,
-             src + (i * source.width()) * pixelsize, width * pixelsize);
-    }
+    software_output_device_->SetActive(painting_, false);
   }
-
-  destination.notifyPixelsChanged();
 }
 
 void OffScreenRenderWidgetHostView::OnPaint(const gfx::Rect& damage_rect,
                                             const SkBitmap& bitmap) {
   TRACE_EVENT0("electron", "OffScreenRenderWidgetHostView::OnPaint");
 
-  HoldResize();
-
   if (parent_callback_) {
     parent_callback_.Run(damage_rect, bitmap);
   } else {
     gfx::Rect damage(damage_rect);
+    gfx::Canvas canvas(size(), 1.0f, false);
```

Comment:
should `1.0f` actually be `scale_factor_` (or `current_device_scale_factor_`...?)

Reply:
I think we are using a `CreateFrom1xBitmap` call to convert the frames to a `NativeImage` instance (effectively), I intended to represent that `1x` in that call. Not sure how it would affect the end result if I called it with the `scale_factor_`, plus we want the frames to end up being the same in size regardless of scale, so the end result should be an unscaled bitmap/canvas.